File: app/src/main/java/com/example/a2ui/chat/domain/model/Message.kt
package com.example.a2ui.chat.domain.model

import androidx.compose.runtime.Immutable
import com.contextable.a2ui4k.model.UiDefinition
import kotlinx.serialization.json.JsonObject

@Immutable
data class Message(
    val id: String,
    val content: String,
    val sender: Sender,
    val timestamp: Long,
    val isLoading: Boolean = false,
    val uiDefinition: UiDefinition? = null,
    /** Reactive data to populate the A2UISurface's DataModel. */
    val dataModelJson: JsonObject? = null
)

File: ios/A2UIChatApp/Domain/Models/Message.swift
import Foundation

struct Message: Identifiable {
    let id: String
    var content: String
    let sender: Sender
    let timestamp: Date
    var isLoading: Bool = false
    var uiDefinition: UiDefinition? = nil
    var dataModelJson: [String: Any]? = nil

    init(
        id: String = UUID().uuidString,
        content: String,
        sender: Sender,
        timestamp: Date = Date(),
        isLoading: Bool = false,
        uiDefinition: UiDefinition? = nil,
        dataModelJson: [String: Any]? = nil
    ) {
        self.id = id
        self.content = content
        self.sender = sender
        self.timestamp = timestamp
        self.isLoading = isLoading
        self.uiDefinition = uiDefinition
        self.dataModelJson = dataModelJson
    }
}
